assets/javascripts/redmine_s3.js

```diff
@@ -0,0 +1,15 @@
+/*
+ * This script handles thumbnail image missing errors
+ */
+
+$(function() {
```

Comment:
So that we are not checking whether thumbnail exists on s3. We just give a link where thumbnail is  supposedly stored without any S3 api calls, and if browser encounters an error (because it can't load the image) - then we make an Ajax call and redmine creates a thumbnail for us.